mcs/class/corlib/System/Delegate.cs

```diff
@@ -499,7 +499,7 @@ public override int GetHashCode ()
 
 			m = Method;
 
-			return (m != null ? m.GetHashCode () : GetType ().GetHashCode ()) ^ (m_target != null ? m_target.GetHashCode () : 0);
+			return (m != null ? m.GetHashCode () : GetType ().GetHashCode ()) ^ (m_target != null ? RuntimeHelpers.GetHashCode (m_target) : 0);
```

Comment:
RuntimeHelpers.GetHashCode handles null so that null check condition is not necessary

Reply:
pushed a fixup